lib/Doctrine/DBAL/TransactionBuilder.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace Doctrine\DBAL;
+
+/**
+ * Builds a transaction with a fluent API.
+ */
+class TransactionBuilder
+{
+    const ISOLATION_LEVEL = 'dbal.isolation-level';
+
+    /**
+     * The transaction manager.
+     *
+     * @var \Doctrine\DBAL\TransactionManager
+     */
+    private $transactionManager;
+
+    /**
+     * The configuration for the transaction to be built.
+     *
+     * @var array
+     */
+    private $configuration = array();
+
+    /**
+     * Class constructor.
+     *
+     * @param TransactionManager $manager
+     */
+    public function __construct(TransactionManager $manager)
+    {
+        $this->transactionManager = $manager;
+    }
+
+    /**
+     * Sets a configuration variable.
+     *
+     * @param string $configurationName  The configuration variable name.
+     * @param string $configurationValue The configuration value.
+     *
+     * @return \Doctrine\DBAL\TransactionBuilder The current instance for chaining.
+     */
+    protected function with($configurationName, $configurationValue)
+    {
+        $this->configuration[$configurationName] = $configurationValue;
```

Comment:
The concept of configuration seems too generic. Why not have as many properties of the needed types as there are configuration parameters? E.g. `$this->isolationLevel` would make more sense to me.

Reply:
This stems from my original idea to have a generic transaction configuration API [that can be reused in the ORM](https://github.com/doctrine/doctrine2/pull/949#issuecomment-53148692). But If we're [not going to expose the DBAL transaction in the ORM](https://github.com/doctrine/doctrine2/pull/949#issuecomment-370002870), then I agree with you, this is much cleaner.